model/jpa/src/main/java/org/keycloak/models/jpa/entities/RoleEntity.java

```diff
@@ -99,6 +98,9 @@ public class RoleEntity {
     @JoinTable(name = "COMPOSITE_ROLE", joinColumns = @JoinColumn(name = "COMPOSITE"), inverseJoinColumns = @JoinColumn(name = "CHILD_ROLE"))
     private Set<RoleEntity> compositeRoles;
 
+    @ManyToMany(mappedBy = "compositeRoles", fetch = FetchType.LAZY, cascade = {})
```

Comment:
@ahus1 Just out of curiosity - We do not need to update liquibase changelog for ManyToMany relationship, do we?

Reply:
@mabartos - this ManyToMany annotation points to the relation above. It is the same table, and the same data (no new data is added). It only adds the necessary Java code to navigate it from child to parent, while the one above allows to navigate from parent to child.

Reply:
@ahus1 Ahhh, nice. Thanks for the explanation.